index.js

```diff
@@ -4368,9 +4620,31 @@ Agent.prototype.request = function (requestMessage, rinfo) {
 					oid: requestPdu.varbinds[i].oid
 				});
 
-				if ( requestPdu.type == PduType.SetRequest ) {
-					mibRequests[i].setType = requestPdu.varbinds[i].type;
-					mibRequests[i].setValue = requestPdu.varbinds[i].value;
+				if ( requestPdu.type == PduType.SetRequest && instanceNode.address.length >= 2 ) {
+					var addrLen = instanceNode.address.length;
+					var row = instanceNode.address[addrLen - 1];
+					var column = instanceNode.address[addrLen - 2];
+					var provider = providerNode.provider;
+					var name = provider.name;
+
+					// Check here for whether this is a RowStatus
+					// column with a value of 6 (delete), and do the
+					// deletion here. (Need to figure out how to
+					// respond and exit early, in that case.)
+					if ( requestPdu.varbind.value === RowStatus["destroy"] &&
+						typeof provider.rowStatusColumn == "number" &&
+						column === provider.rowStatusColumn ) {
+
+						// Delete the table row
+						this.mib.deleteTableRow ( name, row );
+
+						//
+						// TODO: What goes in mibRequests[i]?????
+						//
```

Comment:
These are meant to reflect the incoming varbind type and value, so I don't think you need to deviate from the existing code there:
```
mibRequests[i].setType = requestPdu.varbinds[i].type;
mibRequests[i].setValue = requestPdu.varbinds[i].value;
```